src/k2/tso/service/TSOController.cpp

```diff
@@ -151,25 +151,26 @@ seastar::future<> TSOService::TSOController::GetAllWorkerURLs()
       2. TSO servers could form a gossip cluster, to discovery each other and more rarely to detect if anyone's 
          TimeAuthority/AtomicClock is out of band. 
 */
-void TSOService::TSOController::RegisterGetTSOServerURLs()
+seastar::future<std::tuple<Status, dto::GetTSOServerURLsResponse>>
+TSOService::TSOController::handleGetTSOServerURLs(dto::GetTSOServerURLsRequest&& request)
 {
-    k2::RPC().registerMessageObserver(dto::Verbs::GET_TSO_SERVER_URLS, [this](k2::Request&& request) mutable
-    {
-        auto response = request.endpoint.newPayload();
-        response->write(_TSOServerURLs);
-        K2LOG_D(log::tsoserver, "returned TSO Server TCP endpoints are: {}", _TSOServerURLs);
-        return k2::RPC().sendReply(std::move(response), request);
-    });
+    K2LOG_D(log::tsoserver, "handleGetTSOServerURLs");
+    (void) request;
+
+    GetTSOServerURLsResponse response{.serverURLs = _TSOServerURLs};
+    K2LOG_D(log::tsoserver, "returned TSO Server TCP endpoints are: {}", _TSOServerURLs);
+    return RPCResponse(Statuses::S200_OK("OK"), std::move(response));
 }
 
-void TSOService::TSOController::RegisterGetTSOServiceNodeURLs()
+seastar::future<std::tuple<Status, dto::GetTSOServiceNodeURLsResponse>>
+TSOService::TSOController::handleGetTSOServiceNodeURLs(dto::GetTSOServiceNodeURLsRequest&& request)
 {
-    k2::RPC().registerMessageObserver(dto::Verbs::GET_TSO_SERVICE_NODE_URLS, [this](k2::Request&& request) mutable
-    {
-        auto response = request.endpoint.newPayload();
-        response->write(_workersURLs);
-        return k2::RPC().sendReply(std::move(response), request);
-    });
+    K2LOG_D(log::tsoserver, "handleGetTSOServiceNodeURLs");
+    (void) request;
+
+    GetTSOServiceNodeURLsResponse response{.serviceNodeURLs = _workersURLs};
+    K2LOG_D(log::tsoserver, "returned TSO service nodes endpoints are: {}", _workersURLs);
+    return RPCResponse(Statuses::S200_OK("OK"), std::move(response));

```

Comment:
no need to specify a custom message unless you really want it in the status. just Statuses::S200_OK() would do too

Reply:
Tried that, didn't work as () operator need a string param. Also tried directly Statuses::S200_OK, it doesn't work either as right value is needed here. Prefer not giving default empty string value to the () string param which will loose caller requirement to be explicit. So, leave as it is.